### ISSUE TITLE
Exception trying to set active layer to a CSV file

### ISSUE THREAD

**Issue by timlinux** [AIFDR/inasafe#1292]
# Problem

I have found some layer types raise an exception in InaSAFE.

# Details

```python
An error has occured while executing Python code:


Traceback (most recent call last):
  File "/home/timlinux/.qgis2/python/plugins/inasafe-dev/safe_qgis/widgets/dock.py", line 768, in get_layers
    self.keyword_io.update_keywords(layer, {'title': name})
  File "/home/timlinux/.qgis2/python/plugins/inasafe-dev/safe_qgis/utilities/keyword_io.py", line 165, in update_keywords
    self.write_keywords(layer, existing_keywords)
  File "/home/timlinux/.qgis2/python/plugins/inasafe-dev/safe_qgis/utilities/keyword_io.py", line 142, in write_keywords
    write_keywords_to_file(source, keywords)
  File "/home/timlinux/.qgis2/python/plugins/inasafe-dev/safe_qgis/safe_interface.py", line 383, in write_keywords_to_file
    safe_write_keywords(keywords, filename)
  File "/home/timlinux/.qgis2/python/plugins/inasafe-dev/safe/storage/utilities.py", line 186, in write_keywords
    handle = file(filename, 'w')
IOError: [Errno 2] No such file or directory: 'file:///home/timlinux/Downloads/GL%20LR%20SL%20Health%20Care%20System%20Public%20-%20Guinea%20Health%20Facilities.keywords'


Python version:
2.7.6 (default, Mar 22 2014, 23:03:41) 
[GCC 4.8.2]


QGIS version:
2.5.0-Brighton Brighton, 4c1c322

Python path: ['/usr/local/qgis-master/share/qgis/python/plugins/processing', '/home/timlinux/.qgis2/python/plugins/QGIS-Gazetteer-Plugin', '/usr/local/qgis-master/share/qgis/python', u'/home/timlinux/.qgis2/python', u'/home/timlinux/.qgis2/python/plugins', '/usr/local/qgis-master/share/qgis/python/plugins', '/usr/local/qgis-master/share/qgis/python', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-x86_64-linux-gnu', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages/PILcompat', '/usr/lib/python2.7/dist-packages/gtk-2.0', '/usr/lib/pymodules/python2.7', '/usr/lib/python2.7/dist-packages/ubuntu-sso-client', '/usr/lib/python2.7/dist-packages/wx-2.8-gtk2-unicode', '/home/timlinux/.qgis2/python/plugins/StreamFeatureExtractor/third_party', '/home/timlinux/.qgis2/python/plugins/inasafe-dev', '/home/timlinux/.qgis2/python/plugins/inasafe-dev/third_party', '/usr/local/qgis-master/share/qgis/python/plugins/fTools/tools']
```

# Proposed solution

@mbernasocchi Probably these keywords should be written to the SQLITE keyword store.

At the very least we should avoid raising an exception

**Comment by ismailsunni** [comment]
I have fixed this in #1515 

But only as far as InaSAFE won't crash when CSV layer is selected

**Comment by akbargumbira** [comment]
Hi, if the provider type of the layer is delimitedtext, I set it to use keywords db (https://github.com/AIFDR/inasafe/pull/1521). I tried making a csv file containing a point. I assigned it as volcano hazard and InaSAFE can run this without problem. 

I am closing this. Please reopen if it's still not working
